platform/logging/__tests__/LogLevel.test.ts

```diff
@@ -0,0 +1,117 @@
+import { LogLevel } from '../LogLevel';
+
+const allLogLevels = [
+  LogLevel.Off,
+  LogLevel.Fatal,
+  LogLevel.Error,
+  LogLevel.Warn,
+  LogLevel.Info,
+  LogLevel.Debug,
+  LogLevel.Trace,
+  LogLevel.All
+];
+
+test('`LogLevel.All` supports all log levels.', () => {
+  for (const level of allLogLevels) {
+    expect(LogLevel.All.supports(level)).toBe(true);
+  }
+});
+
+test('`LogLevel.Trace` supports `Trace, Debug, Info, Warn, Error, Fatal, Off`.', () => {
+  const supportedLevels = [
+    LogLevel.Off,
+    LogLevel.Fatal,
+    LogLevel.Error,
+    LogLevel.Warn,
+    LogLevel.Info,
+    LogLevel.Debug,
+    LogLevel.Trace
+  ];
+
+  for (const level of allLogLevels) {
+    expect(LogLevel.Trace.supports(level)).toBe(
+      supportedLevels.includes(level)
+    );
+  }
+});
+
+test('`LogLevel.Debug` supports `Debug, Info, Warn, Error, Fatal, Off`.', () => {
+  const supportedLevels = [
+    LogLevel.Off,
+    LogLevel.Fatal,
+    LogLevel.Error,
+    LogLevel.Warn,
+    LogLevel.Info,
+    LogLevel.Debug
+  ];
+
+  for (const level of allLogLevels) {
+    expect(LogLevel.Debug.supports(level)).toBe(
+      supportedLevels.includes(level)
```

Comment:
Ah, damn, read the code wrong. Thought it looped through the supported levels, not all. Never been a fan of tests with `for` and `if` (easy to have a bug in both the implementation and the test, or to assert on something that's not what you think it is), but it can just stay as-is